lib/spack/spack/test/variant.py

```diff
@@ -734,3 +734,38 @@ def test_conditional_value_comparable_to_bool(other):
     value = spack.variant.Value("98", when="@1.0")
     comparison = value == other
     assert comparison is False
+
+
+@pytest.mark.regression("40405")
+def test_wild_card_valued_variants_equivalent_to_str():
```

Comment:
(minor) Can you add a brief docstring explaining what is tested here (mainly that we can use `*` in directives as an alias to `str`)?